vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/loader/LocalJavaProjectLoader.java

```diff
@@ -92,17 +99,45 @@ public Optional<IJavaProject> load(String uri, Map<Event, Consumer<String>> hand
         }, VividusStudioException::new);
     }
 
-    private static void build(File projectFolder, Optional<Consumer<String>> messageConsumer) throws Exception
+    private static void build(File projectFolder, Optional<Consumer<String>> buildMessageConsumer,
+            Optional<Consumer<String>> errorMessageConsumer) throws Exception
     {
+        Path propertiesFile = projectFolder.toPath().resolve("gradle.properties");
+        String buildSystemVersion = findProperty(propertiesFile, "buildSystemVersion", true, errorMessageConsumer);
+
+        String buildSystemHome = System.getenv("VIVIDUS_BUILD_SYSTEM_HOME");
+        if (buildSystemHome == null)
+        {
+            String buildSystemRootDir = findProperty(propertiesFile, "buildSystemRootDir", false, errorMessageConsumer);
+            buildSystemHome = projectFolder.toPath().resolve(buildSystemRootDir != null ? buildSystemRootDir : EMPTY)
+                    .toString();
+        }
+
+        Path buildSystem = Paths.get(buildSystemHome, buildSystemVersion);
+        if (!Files.exists(buildSystem))
+        {
+            String message = "Neither environment variable \"VIVIDUS_BUILD_SYSTEM_HOME\" is set nor"
+                    + " embedded build system is synced. Please check the build system guide at "
+                    + "https://github.com/vividus-framework/vividus-build-system or clone this repo "
+                    + "recursively: git clone --recursive <git-repository-url>";
+            errorMessageConsumer.ifPresent(c -> c.accept(message));
+            throw new BuildException(message);
+        }
+
+        Path wrapper = buildSystem.resolve("gradle").resolve("wrapper").resolve("gradle-wrapper.properties");
+        String distribution = findProperty(wrapper, "distributionUrl", true, errorMessageConsumer);
```

Comment:
one more place to maintain build system resolution logic, what about just running:
```
./gradlew -version   
```
and checking the version or printing the error?

Reply:
- this logic is covered with unit tests
- gradlew -version will still lead to the download and usage of invalid gradle version

Reply:
nvm